### ISSUE TITLE
Hyperdrive for movielens with one algo

### ISSUE THREAD

**Issue by anargyri** [Microsoft/Recommenders#297]
### *What* is affected by this?
Benchmarking results in Readme and notebooks for the algos.

### Expected behavior (i.e. solution)
Get optimal hyperparameters for all algos by exhaustive grid search. Do this only for Movielens100K and 1M in order to limit computation time.

**Comment by anargyri** [comment]
@yueguoguo has made some good points in the discussion today:
- The various methodologies for hyperparameter tuning may not be well adapted to recommendations problems when we wish to optimize ranking metrics, especially to the loss involved in Bayesian personalized ranking. 
- Hyperdrive uses multi-arm bandits, which are more online in spirit. Bayesian optimization approaches also use exploration - exploitation.
- There are some researchers working on optimizing BPR loss in the case of ranking metrics.  

My comments are that: 
- We have not tried BPR to know how scalable and efficient it is. Spotlight has an implementation https://maciejkula.github.io/spotlight/losses.html?highlight=bpr#spotlight.losses.bpr_loss but this is in Pytorch, and we need to check efficiency first. It also applies only to matrix factorization. I haven't found any BPR implementation on Spark.
- It seems that Hyperdrive can work as a black box and so, we can pass to it a python script that uses e.g. the Spotlight code above. Of course, hyperparam search will not be informed by the nature of the method. One advantage of Hyperdrive that I see is the parallelization is done automatically, which would probably not be the case with a third-party method from academia. I.e. they have done the dirty work for us.
- If we implement ourselves a regression based approach for tuning it could in fact work better than bandits or Bayesian tuning. However we would need to also implement the parallelization and would lose the benefits offered by AML.
